components/dialog/RotateEncryptionKeyDialog.vue

```diff
@@ -24,27 +26,53 @@ export default {
   },
 
   async fetch() {
-    const c = this.cluster;
+    if (!this.$store.getters['isRancher']) {
+      // This fetch function is getting snapshots, which are associated
+      // with cluster manager. Rancher Desktop doesn't come with cluster
+      // manager, so for Rancher Desktop we return nothing.
+      return [];
+    }
 
-    if ( c.isRke1 && this.$store.getters['isRancher'] ) {
-      let etcdBackups = await this.$store.dispatch('rancher/findAll', { type: NORMAN.ETCD_BACKUP });
+    const getEtcdBackups = async() => {
+      if ( this.cluster.isRke1) {
+        let etcdBackups = await this.$store.dispatch('rancher/findAll', { type: NORMAN.ETCD_BACKUP });
 
-      etcdBackups = etcdBackups.filter(backup => backup.clusterId === c.metadata.name);
+        etcdBackups = etcdBackups.filter(backup => backup.clusterId === this.cluster.metadata.name);
 
-      const backups = sortBy(etcdBackups, ['created']).reverse();
+        return etcdBackups;
+      }
+
+      if (this.cluster.isRke2) {
+        let etcdBackups = await this.$store.dispatch('management/findAll', { type: SNAPSHOT });
+
+        etcdBackups = etcdBackups.filter(backup => backup.clusterId === this.cluster.id);
+
+        return etcdBackups;
+      }
+
+      return [];
+    };
+
+    const etcdBackups = await getEtcdBackups();
+
+    const getFormattedCreatedDate = (createdDate) => {
```

Comment:
Maybe we can take also this one out from `fetch`.

Reply:
Moved it into `methods`